aixplain/factories/finetune_factory.py

```diff
@@ -0,0 +1,96 @@
+from aixplain.modules.dataset import Dataset
+from aixplain.modules.model import Model
+from aixplain.modules.finetune import Finetune
+from aixplain.modules.cost import Cost
+from aixplain.utils import config
+from aixplain.utils.file_utils import _request_with_retry
+from urllib.parse import urljoin
+from warnings import warn
+
+
+class FinetuneFactory:
+    """A static class for creating and managing the FineTune experience.
+
+    Attributes:
+        api_key (str): The TEAM API key used for authentication.
+        backend_url (str): The URL for the backend.
+    """
+
+    api_key = config.TEAM_API_KEY
+    aixplain_key = config.AIXPLAIN_API_KEY
+    backend_url = config.BACKEND_URL
+
+    @classmethod
+    def _create_cost_from_response(cls, response: Dict) -> Cost:
+        """Create a Cost object from the response dictionary.
+
+        Args:
+            response (Dict): The response dictionary containing cost information.
+
+        Returns:
+            Cost: The Cost object created from the response.
+        """
+        return Cost(response["trainingCost"], response["inferenceCost"], response["hostingCost"])
+
+    @classmethod
+    def create(
+        cls, name: Text, dataset_list: List[Dataset], model: Model, train_percentage: float = 100, dev_percentage: float = 0
+    ) -> Finetune:
+        """Create a Finetune object with the provided information.
+
+        Args:
+            name (Text): Name of the Finetune.
+            dataset_list (List[Dataset]): List of Datasets to be used for fine-tuning.
+            model (Model): Model to be fine-tuned.
+            train_percentage (float, optional): Percentage of training samples. Defaults to 100.
+            dev_percentage (float, optional): Percentage of development samples. Defaults to 0.
+
+        Returns:
+            Finetune: The Finetune object created with the provided information or None if there was an error.
+        """
+        payload = {}
+        try:
+            url = urljoin(cls.backend_url, f"sdk/finetune/cost-estimation")
+            headers = {"Authorization": f"Token {cls.api_key}", "Content-Type": "application/json"}
+            payload = json.dumps(
+                {
+                    "datasets": [{"datasetId": dataset.id, "trainPercentage": train_percentage, "devPercentage": dev_percentage} for dataset in dataset_list],
+                    "sourceModelId": model.id,
+                }
+            )
+            print(f"Payload: {payload}")
+            r = _request_with_retry("post", url, headers=headers, data=payload)
+            resp = r.json()
+            logging.info(f"Creating Finetune Job: Status for: {resp}")
+            cost = cls._create_cost_from_response(resp)
+            return Finetune(name, dataset_list, model, cost, train_percentage=train_percentage, dev_percentage=dev_percentage)
+        except Exception as e:
+            error_message = f"Estimating cost: Error in cost estimation with payload {payload} : {e}"
```

Comment:
Are all exceptions cost estimation exceptions?
It would be better to get back the entire traceback with logging.exception in the following line.